src/TestProblem/Hydro/ClusterMerger/Par_Init_ByFunction_ClusterMerger.cpp

```diff
@@ -291,6 +295,57 @@ void Par_Init_ByFunction_ClusterMerger( const long NPar_ThisRank, const long NPa
 
    if ( MPI_Rank == 0 )    Aux_Message( stdout, "done\n" );
 
+
+   // label cluster centers
+   if ( Merger_Coll_LabelCenter ) {
+      if ( MPI_Rank == 0 )    Aux_Message( stdout, "   Labeling cluster centers ... " );
+
+      const double Centers[3][3] = {  { ClusterCenter1[0], ClusterCenter1[1], ClusterCenter1[2] },
+                                      { ClusterCenter2[0], ClusterCenter2[1], ClusterCenter2[2] },
+                                      { ClusterCenter3[0], ClusterCenter3[1], ClusterCenter3[2] }  };
+      long pidx_offset = 0;
+
+      for (int c=0; c<NCluster; c++) {
+         long   min_pidx   = -1;
+         real   min_pos[3] = { NULL_REAL, NULL_REAL, NULL_REAL };
+         double min_r      = __DBL_MAX__;
+
+         // get the particle in this rank closest to the cluster center
+         for (long p=pidx_offset; p<pidx_offset+NPar_ThisRank_EachCluster[c]; p++) {
+            const double r = SQR( ParPos[0][p] - Centers[c][0] ) +
+                             SQR( ParPos[1][p] - Centers[c][1] ) +
+                             SQR( ParPos[2][p] - Centers[c][2] );
+            if ( r < min_r ) {
+               min_pidx   = p;
+               min_r      = r;
+               min_pos[0] = ParPos[0][p];
+               min_pos[1] = ParPos[1][p];
+               min_pos[2] = ParPos[2][p];
+            }
+         }
+
+         // collect data among all ranks
+         double min_r_allrank;
+         int    NFound_ThisRank=0, NFound_AllRank;
+         MPI_Allreduce( &min_r, &min_r_allrank, 1, MPI_DOUBLE, MPI_MIN, MPI_COMM_WORLD );
```

Comment:
I don't quite understand this function `MPI_Allreduce`

Reply:
When you run GAMER in parallel, each CPU only owns part of the simulation data, and `MPI` is used to exchange these data among different CPUs. In this case, each CPU first computes its own `min_r`, and then we call `MPI_Allreduce()` to get the minimum of them (with `MPI_MIN`).

I don't think you need to learn `MPI` at this point. Getting some basic ideas of parallel computing should be enough 😁

Reply:
Got it. Thank you!